firebase-common/src/main/java/com/google/firebase/logger/Logger.kt

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.logger
+
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Common logger interface that handles Android logcat logging for Firebase SDKs.
+ *
+ * @hide
+ */
+sealed class Logger
```

Comment:
While this is not part of the dev-visible public API, it's part of our inter-SDK API, and using `sealed` classes can be problematic in APIs

https://g3doc.corp.google.com/firebase/acore/g3doc/docs/kotlin_best_practices.md?cl=head#api-surface

http://goto.google.com/androidx-api-guidelines#exhaustive-when

Do we envision this class gaining more subclasses anytime soon?

Reply:
I was thinking there would be logic using a `when` clause for a `FakeLogger` vs `AndroidLogger` but I ended up never needing that. So I'll remove the `sealed` since it has no purpose right now and could be problematic in the future.